scripts: add `drtprod` wrapper for managing drt clusters

dt opened the pull request:
Release note: none.
Epic: none.

dt committed:
scripts: add wrapper for managing drt clusters

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120056)
<!-- Reviewable:end -->

dt commented:
> why we need the wrapper instead of exporting the env vars locally

I think it is likely that someone will forget one vars -- and we do have to pass both and maybe more to come -- or accidentally pass it when they do not mean to (e.g. one reason we want a separate project is to prevent `destroy --mine` getting run here, but if bash history that passes the var is floating around that becomes possible).

> `bin` an unusual choice for the location, ...  scripts/ for this purpose.

We put `roachprod` in `bin` when `dev` builds it, and this is essentially an alternative version of roachprod for managing drt clusters rather than roachrod clusters. If, instead of configuring via env vars I'd chosen to have a go wrapper around cmd/roachprod that overrode a few package vars and spit out a binary, that would go in `bin`, not `scripts`.

I'm fine moving it to `scripts` though and updating my PATH if that seems preferred.

renatolabs commented:
Thanks for explaining.

Re: location, the `scripts` folder already has a number of scripts that could be described in the same way (wrappers for other binaries that would live in `bin` if they were Go packages compiled by `dev`), and I don't see a strong reason for this script in particular to break that convention.

But it's pretty inconsequential so if I'm the only one of this opinion, it's fine.

dt commented:
I've moved it to `scripts/` and added commentary to the file explaining why it exists. 

While I'm here, I've also expanded the script slightly to validate arguments match the requirements for DRT (secure mode and static ports), which I think helps show the value of the wrapper.

herkolategan commented on scripts/drtprod:
```diff
@@ -0,0 +1,19 @@
+    echo "--secure is required when starting DRT clusters"
+    exit 1
+  fi
+  if [[ "$*" != *"--sql-port 26257"* ]]; then
```
Thanks for adding this, I added some additional checks in `roachprod` to prevent using DNS/services on clusters that do not support it.

dt commented:
TFTR!

bors r+
